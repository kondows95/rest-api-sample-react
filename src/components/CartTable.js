import React from 'react';
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import { BASEURL_ITEM_IMAGES } from '../constants'
import QuantitySelect from '../containers/QuantitySelect'
import CartItemPropTypes from './CartItemPropTypes'
import {Box, Grid, Paper, CardMedia, Divider, Hidden, Link} from '@material-ui/core'


const useStyles = makeStyles(theme => ({
  gridRight: {
    margin: 'auto',
    textAlign: 'right',
    paddingTop: theme.spacing(1),
    paddingBottom: theme.spacing(1),
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
  },
  gridLeft: {
    margin: 'auto',
    textAlign: 'left',
    paddingTop: theme.spacing(1),
    paddingBottom: theme.spacing(1),
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
  },
  descContainer: {
    display: 'flex',
  },
  descImagePC: {
    width: 50,
    height: 64
  },
  descImageMobile: {
    height: 0,
    paddingTop: '128%',
  },
  descTitle: {
    display: 'flex',
    flexDirection: 'column',
    margin: 'auto',
    marginLeft: theme.spacing(2),
  },
  deleteLink: {
    marginTop: theme.spacing(1),
  },
  rightItem: {
    paddingLeft: theme.spacing(1),
  }
}));

const CartTable = ({ cart, deleteCartItem }) => {
  const classes = useStyles()

  //componet list for PC
  const tableRows = [] //for PC
  tableRows.push(
    <Grid key={0} container>
      <Grid item xs={4}>
        <Box textAlign="left" my={2} ml={2}>Desc</Box>
      </Grid>
      <Grid item xs={2}>
        <Box textAlign="right" my={2}>@ (Ks)</Box>
      </Grid>
      <Grid item xs={2}>
        <Box textAlign="right" my={2}>Qty.</Box>
      </Grid>
      <Grid item xs={2}>
        <Box textAlign="right" my={2} mr={2}>Price (Ks)</Box>
      </Grid>
      <Grid item xs={2}>
        <Box m={1} />
      </Grid>
      <Grid item xs={12}><Divider /></Grid>
    </Grid>
  )

  //componet list for Mobile
  const paperRows = [] //for Mobile

  //loop per cartItem
  cart.forEach((row, i) => {
    const divider = (i + 1 < cart.length) ? (<Grid item xs={12}><Divider /></Grid>) : null

    //rows for PC
    tableRows.push(
      <Grid container key={row.id}>
        <Grid item xs={4}>
          <Box display="flex" flexDirection="row" pl={2} py={1}>
            <CardMedia
              className={classes.descImagePC}
              image={BASEURL_ITEM_IMAGES+row.image}
              //image={"http://127.0.0.1/dummyImage.jpg"}
              title={row.name}
            />
            <Box textAlign="left" ml={2} my="auto" fontWeight={600}>
              {row.name}
            </Box>  
          </Box>
          
        </Grid>
        <Grid item xs={2} >
          <Box display="flex" height="100%">
            <Box textAlign="right" my="auto" ml="auto" mr={0}>
              {row.price}
            </Box>  
          </Box>   
        </Grid>
        <Grid item xs={2}>
          <Box display="flex" height="100%">
            <Box my="auto" ml="auto" mr={0}>
              <Box pl="auto">
                <QuantitySelect id={row.id} quantity={row.quantity} />
              </Box>
            </Box>
          </Box>
        </Grid>
        <Grid item xs={2} className={classes.gridRight}>
          <Box textAlign="right" fontWeight={600}>
            {row.subTotal}
          </Box>
        </Grid>
        <Grid item xs={2} className={classes.gridRight}>
          <Link onClick={() => deleteCartItem(row.id)}>
            Delete
          </Link>
        </Grid>
        {divider}
      </Grid>
    )

    //rows for Mobile
    paperRows.push(
      <Grid container key={row.id}>
        <Grid item xs={4}>
          <Box m={2}>
            <CardMedia
              className={classes.descImageMobile}
              //image={BASEURL_ITEM_IMAGES+row.image}
              image={"http://127.0.0.1/dummyImage.jpg"}
              title={row.name}
            />
          </Box>
        </Grid>
        <Grid item xs={8}>
          <Box display="flex" height='100%'>
            <Box flexGrow="1" my="auto">
              <Box fontWeight={600} my={1} textAlign="left" >
                {row.name}
              </Box>
              <Box display="flex" my={1} flexDirection="row">
                <Box fontWeight={600} textAlign="left" >
                  {row.subTotal} Ks
                </Box>
                <Box ml="auto" mr={2}>
                  <Link onClick={() => deleteCartItem(row.id)}>
                    Delete
                  </Link>
                </Box>
              </Box>
              <Box display="flex" my={1} flexDirection="row">
                <QuantitySelect id={row.id} quantity={row.quantity} />
                <Box textAlign="left">
                  @{row.price} Ks
                </Box>
              </Box>
            </Box>
          </Box>
        </Grid>
        <Grid item xs={12}>{divider}</Grid> 
      </Grid>
    )
  })

  return (
    <Paper>
      <Hidden smUp implementation="css">
      {paperRows}  
      </Hidden>
      <Hidden xsDown implementation="css">
        {tableRows}
      </Hidden>
    </Paper>
  )
}

CartTable.propTypes = {
  cart: PropTypes.arrayOf(CartItemPropTypes.isRequired).isRequired,
}

export default CartTable;
