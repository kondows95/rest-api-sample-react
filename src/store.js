import rootReducer from './modules/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
export default createStore(rootReducer, applyMiddleware(thunk));