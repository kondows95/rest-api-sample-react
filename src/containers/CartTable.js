import { connect } from 'react-redux'
import CartTable from '../components/CartTable'
import { deleteCartItem } from '../modules/cart.ts'

export default connect(
  (state) => ({
    cart: state.cart.rows
  }),
  (dispatch) => ({
    deleteCartItem: (itemId) => dispatch(deleteCartItem(itemId)),
  })
)(CartTable)
