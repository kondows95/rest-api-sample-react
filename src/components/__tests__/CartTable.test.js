import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import CartTable from '../CartTable';
import { Provider } from 'react-redux'
import store from '../../store';
import {  BrowserRouter as Router } from 'react-router-dom';
let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})

const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;

describe("CartTable component", () => {
  let cart = [{ id: 1, name: "a", quantity: 1, price: 1000 },
  { id: 2, name: "b", quantity: 1, price: 1000 }];
  it('matches the snapshot', () => {
    const CartTableSnapshot = renderer.create(<Provider store={store}>
      <Router>
        <CartTable cart={cart} />
      </Router>
    </Provider>).toJSON();
    expect(CartTableSnapshot).toMatchSnapshot();
  })
})

describe("testing cart table", () => {
  it('should exists one create button and one name textfield', () => {
    let cart = [
      { id: 1, name: "a", quantity: 1, price: 1000, subTotal: 1000 },
      { id: 2, name: "b", quantity: 1, price: 1000, subTotal: 1000 }
    ];
    const deleteCartItem = jest.fn((id) => {
      if (id === 1) cart = [{ id: 2, name: "b", quantity: 1, price: 1000, subTotal: 1000 }];
      else cart = [{ id: 1, name: "a", quantity: 1, price: 1000, subTotal: 1000 }];
    });
    act(() => {
      ReactDOM.render((<Provider store={store}>
        <Router>
          <CartTable cart={cart} deleteCartItem={deleteCartItem} />
        </Router>
      </Provider>), container);
    });
    const linkArr = document.querySelectorAll('a')
    expect(linkArr[0].textContent.toUpperCase()).toBe("DELETE");
    act(() => {
      linkArr[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    const expectedCart =[{ id: 2, name: "b", quantity: 1, price: 1000, subTotal: 1000 }];
    expect(deleteCartItem).toHaveBeenCalled();
    expect(cart).toStrictEqual(expectedCart);
  });

})




