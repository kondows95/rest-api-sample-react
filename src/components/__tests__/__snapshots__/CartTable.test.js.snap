// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTable component matches the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="PrivateHiddenCss-smUp-40"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="MuiBox-root MuiBox-root-155"
        >
          <div
            className="MuiCardMedia-root makeStyles-descImageMobile-5"
            style={
              Object {
                "backgroundImage": "url(\\"http://mb-amplify-sampledc98ea5760ad43938d826ce868526e9d-dev.s3-ap-northeast-1.amazonaws.com/public/undefined\\")",
              }
            }
            title="a"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <div
          className="MuiBox-root MuiBox-root-158"
        >
          <div
            className="MuiBox-root MuiBox-root-159"
          >
            <div
              className="MuiBox-root MuiBox-root-160"
            >
              a
            </div>
            <div
              className="MuiBox-root MuiBox-root-161"
            >
              <div
                className="MuiBox-root MuiBox-root-162"
              >
                 Ks
              </div>
              <div
                className="MuiBox-root MuiBox-root-163"
              >
                <a
                  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                >
                  Delete
                </a>
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-200"
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline"
                onClick={[Function]}
              >
                <select
                  className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  value={1}
                >
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                  <option
                    value={3}
                  >
                    3
                  </option>
                  <option
                    value={4}
                  >
                    4
                  </option>
                  <option
                    value={5}
                  >
                    5
                  </option>
                  <option
                    value={6}
                  >
                    6
                  </option>
                  <option
                    value={7}
                  >
                    7
                  </option>
                  <option
                    value={8}
                  >
                    8
                  </option>
                  <option
                    value={9}
                  >
                    9
                  </option>
                  <option
                    value={10}
                  >
                    10
                  </option>
                  <option
                    value={11}
                  >
                    11
                  </option>
                  <option
                    value={12}
                  >
                    12
                  </option>
                  <option
                    value={13}
                  >
                    13
                  </option>
                  <option
                    value={14}
                  >
                    14
                  </option>
                  <option
                    value={15}
                  >
                    15
                  </option>
                  <option
                    value={16}
                  >
                    16
                  </option>
                  <option
                    value={17}
                  >
                    17
                  </option>
                  <option
                    value={18}
                  >
                    18
                  </option>
                  <option
                    value={19}
                  >
                    19
                  </option>
                  <option
                    value={20}
                  >
                    20
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiNativeSelect-icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <div
                className="MuiBox-root MuiBox-root-247"
              >
                @
                1000
                 Ks
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <hr
            className="MuiDivider-root"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="MuiBox-root MuiBox-root-253"
        >
          <div
            className="MuiCardMedia-root makeStyles-descImageMobile-5"
            style={
              Object {
                "backgroundImage": "url(\\"http://mb-amplify-sampledc98ea5760ad43938d826ce868526e9d-dev.s3-ap-northeast-1.amazonaws.com/public/undefined\\")",
              }
            }
            title="b"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <div
          className="MuiBox-root MuiBox-root-254"
        >
          <div
            className="MuiBox-root MuiBox-root-255"
          >
            <div
              className="MuiBox-root MuiBox-root-256"
            >
              b
            </div>
            <div
              className="MuiBox-root MuiBox-root-257"
            >
              <div
                className="MuiBox-root MuiBox-root-258"
              >
                 Ks
              </div>
              <div
                className="MuiBox-root MuiBox-root-259"
              >
                <a
                  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                >
                  Delete
                </a>
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-260"
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline"
                onClick={[Function]}
              >
                <select
                  className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  value={1}
                >
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                  <option
                    value={3}
                  >
                    3
                  </option>
                  <option
                    value={4}
                  >
                    4
                  </option>
                  <option
                    value={5}
                  >
                    5
                  </option>
                  <option
                    value={6}
                  >
                    6
                  </option>
                  <option
                    value={7}
                  >
                    7
                  </option>
                  <option
                    value={8}
                  >
                    8
                  </option>
                  <option
                    value={9}
                  >
                    9
                  </option>
                  <option
                    value={10}
                  >
                    10
                  </option>
                  <option
                    value={11}
                  >
                    11
                  </option>
                  <option
                    value={12}
                  >
                    12
                  </option>
                  <option
                    value={13}
                  >
                    13
                  </option>
                  <option
                    value={14}
                  >
                    14
                  </option>
                  <option
                    value={15}
                  >
                    15
                  </option>
                  <option
                    value={16}
                  >
                    16
                  </option>
                  <option
                    value={17}
                  >
                    17
                  </option>
                  <option
                    value={18}
                  >
                    18
                  </option>
                  <option
                    value={19}
                  >
                    19
                  </option>
                  <option
                    value={20}
                  >
                    20
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiNativeSelect-icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <div
                className="MuiBox-root MuiBox-root-261"
              >
                @
                1000
                 Ks
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      />
    </div>
  </div>
  <div
    className="PrivateHiddenCss-xsDown-38"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="MuiBox-root MuiBox-root-262"
        >
          Desc
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-263"
        >
          @ (Ks)
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-264"
        >
          Qty.
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-265"
        >
          Price (Ks)
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-266"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="MuiBox-root MuiBox-root-267"
        >
          <div
            className="MuiCardMedia-root makeStyles-descImagePC-4"
            style={
              Object {
                "backgroundImage": "url(\\"http://mb-amplify-sampledc98ea5760ad43938d826ce868526e9d-dev.s3-ap-northeast-1.amazonaws.com/public/undefined\\")",
              }
            }
            title="a"
          />
          <div
            className="MuiBox-root MuiBox-root-268"
          >
            a
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-269"
        >
          <div
            className="MuiBox-root MuiBox-root-270"
          >
            1000
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-271"
        >
          <div
            className="MuiBox-root MuiBox-root-272"
          >
            <div
              className="MuiBox-root MuiBox-root-273"
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline"
                onClick={[Function]}
              >
                <select
                  className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  value={1}
                >
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                  <option
                    value={3}
                  >
                    3
                  </option>
                  <option
                    value={4}
                  >
                    4
                  </option>
                  <option
                    value={5}
                  >
                    5
                  </option>
                  <option
                    value={6}
                  >
                    6
                  </option>
                  <option
                    value={7}
                  >
                    7
                  </option>
                  <option
                    value={8}
                  >
                    8
                  </option>
                  <option
                    value={9}
                  >
                    9
                  </option>
                  <option
                    value={10}
                  >
                    10
                  </option>
                  <option
                    value={11}
                  >
                    11
                  </option>
                  <option
                    value={12}
                  >
                    12
                  </option>
                  <option
                    value={13}
                  >
                    13
                  </option>
                  <option
                    value={14}
                  >
                    14
                  </option>
                  <option
                    value={15}
                  >
                    15
                  </option>
                  <option
                    value={16}
                  >
                    16
                  </option>
                  <option
                    value={17}
                  >
                    17
                  </option>
                  <option
                    value={18}
                  >
                    18
                  </option>
                  <option
                    value={19}
                  >
                    19
                  </option>
                  <option
                    value={20}
                  >
                    20
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiNativeSelect-icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-gridRight-1 MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-274"
        />
      </div>
      <div
        className="MuiGrid-root makeStyles-gridRight-1 MuiGrid-item MuiGrid-grid-xs-2"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Delete
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="MuiBox-root MuiBox-root-275"
        >
          <div
            className="MuiCardMedia-root makeStyles-descImagePC-4"
            style={
              Object {
                "backgroundImage": "url(\\"http://mb-amplify-sampledc98ea5760ad43938d826ce868526e9d-dev.s3-ap-northeast-1.amazonaws.com/public/undefined\\")",
              }
            }
            title="b"
          />
          <div
            className="MuiBox-root MuiBox-root-276"
          >
            b
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-277"
        >
          <div
            className="MuiBox-root MuiBox-root-278"
          >
            1000
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-279"
        >
          <div
            className="MuiBox-root MuiBox-root-280"
          >
            <div
              className="MuiBox-root MuiBox-root-281"
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline"
                onClick={[Function]}
              >
                <select
                  className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  value={1}
                >
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                  <option
                    value={3}
                  >
                    3
                  </option>
                  <option
                    value={4}
                  >
                    4
                  </option>
                  <option
                    value={5}
                  >
                    5
                  </option>
                  <option
                    value={6}
                  >
                    6
                  </option>
                  <option
                    value={7}
                  >
                    7
                  </option>
                  <option
                    value={8}
                  >
                    8
                  </option>
                  <option
                    value={9}
                  >
                    9
                  </option>
                  <option
                    value={10}
                  >
                    10
                  </option>
                  <option
                    value={11}
                  >
                    11
                  </option>
                  <option
                    value={12}
                  >
                    12
                  </option>
                  <option
                    value={13}
                  >
                    13
                  </option>
                  <option
                    value={14}
                  >
                    14
                  </option>
                  <option
                    value={15}
                  >
                    15
                  </option>
                  <option
                    value={16}
                  >
                    16
                  </option>
                  <option
                    value={17}
                  >
                    17
                  </option>
                  <option
                    value={18}
                  >
                    18
                  </option>
                  <option
                    value={19}
                  >
                    19
                  </option>
                  <option
                    value={20}
                  >
                    20
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiNativeSelect-icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-gridRight-1 MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-282"
        />
      </div>
      <div
        className="MuiGrid-root makeStyles-gridRight-1 MuiGrid-item MuiGrid-grid-xs-2"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Delete
        </a>
      </div>
    </div>
  </div>
</div>
`;
